Work-in-progress: use Guzzle 6

darrenmothersele opened the pull request:
This allows use of Guzzle v6. Doesn't support attachments yet.

stof commented on lib/Stampie/Adapter/Guzzle6.php:
```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Created by PhpStorm.
```
this header is wrong

stof commented on lib/Stampie/Adapter/Guzzle6.php:
```diff
@@ -0,0 +1,53 @@
+class Guzzle6 implements AdapterInterface
+{
+    /** @var ClientInterface $client */
+    protected $client;
```
please use the private visibiblity

stof commented:
Cannot be accepted without tests and an a complete implementation

stof commented on lib/Stampie/Adapter/Guzzle6.php:
```diff
@@ -0,0 +1,53 @@
+     * Guzzle6 constructor.
+     * @param $client
+     */
+    public function __construct($client)
```
argument should be typehinted

stof commented on lib/Stampie/Adapter/Guzzle6.php:
```diff
@@ -0,0 +1,53 @@
+    protected $client;
+
+    /**
+     * Guzzle6 constructor.
```
this description is useless. We already know it is a constructor thanks to the method name

Ninir commented:
Really curious to look into this! :-) keep going @darrenmothersele

darrenmothersele commented:
I've fixed code style. And added test coverage for Guzzle 6 Adapter.

I'm just not sure of recommended way to add attachments to PSR-7 request is. I need to look into this.

stof commented on composer.json:
```diff
@@ -30,7 +30,7 @@
         "kriswallsmith/buzz" : "~0.8",
         "guzzle/http"        : "~3",
-        "guzzlehttp/guzzle"  : "~5.2"
+        "guzzlehttp/guzzle": "^6.1"
```
it should allow both actually (and we should adapt the Travis setup to run tests against both)

darrenmothersele replied:
How do I include both versions 5 and 6 of Guzzle?

incarnate replied:
Something like this will allow the range of versions, Composer doesn't allow more than one version of the same package to be installed at once:
`"guzzlehttp/guzzle": ">=5.2 <6.2.0"`

See [this Travis doco](https://docs.travis-ci.com/user/languages/php#Testing-Against-Multiple-Versions-of-Dependencies-(e.g.-Symfony)) to configure testing against multiple dependency versions.

stof commented on tests/Stampie/Tests/Adapter/Guzzle6Test.php:
```diff
@@ -0,0 +1,88 @@
+
+    public function setUp()
+    {
+        if (!interface_exists('GuzzleHttp\ClientInterface')) {
```
not enough to distinguish guzzle 5 and guzzle 6 AFAIK

darrenmothersele replied:
Yeah, that was copied from the v5 test. I added an extra check for v6 below it.

stof commented on tests/Stampie/Tests/Adapter/Guzzle6Test.php:
```diff
@@ -0,0 +1,88 @@
+    public function testAccessibility()
+    {
+        $adapter = new Guzzle6($this->client);
+        $this->assertEquals($this->client, $adapter->getClient());
```
please use ``assertSame``

darrenmothersele replied:
Yes, will change, this was borrowed from the v5 test.

henrikbjorn commented:
1.0.0-alpha1 supports Guzzle6 through HTTPlug
